patches_for_WebRTC_org/m80/0007.2-Properly-handling-async-model-for-initializing-Media.patch

```diff
@@ -20,13 +20,93 @@ device_info_winrt, repent for the goto sins
 No more goto or THR sins
 
 Not using ReleaseInterface anymore
+
+Passing YUV strides and gap down to libyuv
 ---
+ modules/video_capture/video_capture_impl.cc   |  28 +-
+ modules/video_capture/video_capture_impl.h    |   5 +-
  .../windows/device_info_winrt.cc              | 402 +++++++------
  .../windows/help_functions_winrt.cc           | 335 +++++++++--
  .../windows/help_functions_winrt.h            | 123 ++--
- .../windows/video_capture_winrt.cc            | 565 ++++++++++++------
- 4 files changed, 937 insertions(+), 488 deletions(-)
+ .../windows/video_capture_winrt.cc            | 562 +++++++++++-------
+ 6 files changed, 943 insertions(+), 512 deletions(-)
 
+diff --git a/modules/video_capture/video_capture_impl.cc b/modules/video_capture/video_capture_impl.cc
+index 9d53a91157..398e41c07b 100644
+--- a/modules/video_capture/video_capture_impl.cc
++++ b/modules/video_capture/video_capture_impl.cc
+@@ -114,7 +114,10 @@ int32_t VideoCaptureImpl::DeliverCapturedFrame(VideoFrame& captureFrame) {
+   return 0;
+ }
+ 
+-int32_t VideoCaptureImpl::IncomingFrame(uint8_t* videoFrame,
++int32_t VideoCaptureImpl::IncomingFrame(uint8_t* plane_y,
++                                        int32_t stride_y,
++                                        uint8_t* plane_uv,
++                                        int32_t stride_uv,
+                                         size_t videoFrameLength,
+                                         const VideoCaptureCapability& frameInfo,
+                                         int64_t captureTime /*=0*/) {
+@@ -125,16 +128,8 @@ int32_t VideoCaptureImpl::IncomingFrame(uint8_t* videoFrame,
+ 
+   TRACE_EVENT1("webrtc", "VC::IncomingFrame", "capture_time", captureTime);
+ 
+-  // Not encoded, convert to I420.
+-  if (frameInfo.videoType != VideoType::kMJPEG &&
+-      CalcBufferSize(frameInfo.videoType, width, abs(height)) !=
+-          videoFrameLength) {
+-    RTC_LOG(LS_ERROR) << "Wrong incoming frame length.";
+-    return -1;
+-  }
+-
+-  int stride_y = width;
+-  int stride_uv = (width + 1) / 2;
++  int dst_stride_y = width;
++  int dst_stride_uv = (width + 1) / 2;
+   int target_width = width;
+   int target_height = abs(height);
+ 
+@@ -156,7 +151,7 @@ int32_t VideoCaptureImpl::IncomingFrame(uint8_t* videoFrame,
+ 
+   // TODO(nisse): Use a pool?
```

Comment:
nit: by all means use a pool! The corresponding WebRTC-UWP code was a noticeable perf issue on HL. See https://github.com/webrtc-uwp/webrtc-apis/commit/ced46e4c94b227514c078e7004ee776df8c350ea for a fix.

Not relevant to this PR though so can be done separately, just a reminder.

Reply:
Reported as #32.